src/helpers/issue-fetching.ts

```diff
@@ -1,99 +1,62 @@
 
-/**
- * Recursively fetches linked issues and processes them, including fetching comments and specifications.
- *
- * @param params - The parameters required to fetch the linked issues, including context and other details.
- * @returns A promise that resolves to an object containing linked issues, specifications, streamlined comments, and seen issue keys.
- */
 export async function recursivelyFetchLinkedIssues(params: FetchParams) {
+  // take a first run at gathering everything we need and package it up
   const { linkedIssues, seen, specAndBodies, streamlinedComments } = await fetchLinkedIssues(params);
+  // build promises and throttle them; this calls handleSpec which is a recursive function potentially to great depth
   const fetchPromises = linkedIssues.map(async (linkedIssue) => await mergeCommentsAndFetchSpec(params, linkedIssue, streamlinedComments, specAndBodies, seen));
   await throttlePromises(fetchPromises, 10);
+  // handle the keys that have been gathered
   const linkedIssuesKeys = linkedIssues.map((issue) => createKey(`${issue.owner}/${issue.repo}/${issue.issueNumber}`));
+  // exhaustive list of unique keys from the first full pass
   const specAndBodyKeys = Array.from(new Set([...Object.keys(specAndBodies), ...Object.keys(streamlinedComments), ...linkedIssuesKeys]));
+  // this fn throttles from within but again, be weary of the rate limit
   await handleSpecAndBodyKeys(specAndBodyKeys, params, dedupeStreamlinedComments(streamlinedComments), seen);
   return { linkedIssues, specAndBodies, streamlinedComments };
 }
 
-/**
- * Fetches linked issues recursively and processes them.
- *
- * @param params - The parameters required to fetch the linked issues, including context and other details.
- * @returns A promise that resolves to an object containing linked issues, specifications, streamlined comments, and seen issue keys.
- */
 export async function fetchLinkedIssues(params: FetchParams) {
   const { comments, issue } = await fetchIssueComments(params);
   if (!issue) {
     return { streamlinedComments: {}, linkedIssues: [], specAndBodies: {}, seen: new Set<string>() };
   }
-  if (!issue.body || !issue.html_url) {
-    throw logger.error("Issue body or URL not found", { issueUrl: issue.html_url });
-  }
 
   if (!params.owner || !params.repo) {
     throw logger.error("Owner or repo not found");
   }
+
   const issueKey = createKey(issue.html_url);
   const [owner, repo, issueNumber] = splitKey(issueKey);
-  const linkedIssues: LinkedIssues[] = [{ body: issue.body, comments, issueNumber: parseInt(issueNumber), owner, repo, url: issue.html_url }];
+  const linkedIssues: LinkedIssues[] = [{ body: issue.body || "", comments, issueNumber: parseInt(issueNumber), owner, repo, url: issue.html_url }];
   const specAndBodies: Record<string, string> = {};
   const seen = new Set<string>([issueKey]);
 
   comments.push({
-    body: issue.body,
+    body: issue.body || "",
```

Comment:
```suggestion
    body: issue.body ?? null,
```

Reply:
empty strings are safe in this context. "" evaluates to null and if added into the context window it's nothing. The alternative means explicitly typing null and then filtering, wasn't sure which was cleaner.

I'll remove them